test/mode/prod/prod.test.js

```diff
@@ -25,6 +25,17 @@ describe('mode flags', () => {
         });
     });
 
+    it('should load a production config when --mode=abcd is passed', done => {
+        const { stderr, stdout } = run(__dirname, ['--mode', 'abcd']);
+        expect(stderr).toContain('invalid value for "mode"');
+        expect(stdout).toBeTruthy();
+        stat(resolve(__dirname, './bin/main.js'), (err, stats) => {
```

Comment:
Here, we also have to test that `production` has been passed correctly. Which means inside your entry file, you'd need to print something based on `process.env.NODE_ENV === "production"` assert that the correct value is inside the output bundle